src/dataframerow/utils.jl

```diff
@@ -450,6 +450,16 @@ function group_rows(df::AbstractDataFrame)
     return RowGroupDict(df, rhashes, gslots, groups, rperm, starts, stops)
 end
 
+# hash of DataFrame rows based on its values
+# so that duplicate rows would have the same hash
+# table columns are passed as a tuple of vectors to ensure type specialization
+rowhash(cols::Tuple{AbstractVector}, r::Int, h::UInt = zero(UInt))::UInt =
+    hash(cols[1][r], h)
+function rowhash(cols::Tuple{Vararg{AbstractVector}}, r::Int, h::UInt = zero(UInt))::UInt
```

Comment:
it is interesting that we do not have this covered by tests. I will add some.

Reply:
it turns out we do not use these functions. I have removed them. @nalimilan - could you please have a look if we would ever need them? (I have never used them)

Reply:
AFAICT one guy removed the last use of `findrow` and `group_rows` in a recent PR. :-D https://github.com/JuliaData/DataFrames.jl/pull/2641

It's weird that coverage didn't notice this. Maybe we only looked at the changed code and not at unrelated parts?

Reply:
There were tests that checked correctness of internal functions.